src/dotnet/UI.Blazor.App/Services/AppPresenceReporter.cs

```diff
@@ -31,30 +34,40 @@ public AppPresenceReporter(Options settings, IServiceProvider services)
         Clocks = Settings.Clocks ?? services.Clocks();
         UserActivityUI = services.GetRequiredService<UserActivityUI>();
         UserPresences = services.GetRequiredService<IUserPresences>();
+        ChatUI = services.GetRequiredService<ChatUI>();
     }
 
 
     protected override async Task RunInternal(CancellationToken cancellationToken)
     {
         var session = await SessionResolver.GetSession(cancellationToken).ConfigureAwait(false);
-        await foreach (var change in UserActivityUI.LastActiveAt.Changes(cancellationToken)) {
-            var lastActiveAt = change.Value;
-            if (Clocks.SystemClock.Now - lastActiveAt < Settings.AwayTimeout)
+        var cState = await Computed.Capture(() => GetAppPresenceState(cancellationToken)).ConfigureAwait(false);
+        await foreach (var change in cState.Changes(cancellationToken).ConfigureAwait(false)) {
+            var (lastActiveAt, isRecording) = change.Value;
+            if (Clocks.SystemClock.Now - lastActiveAt < Settings.AwayTimeout || isRecording)
```

Comment:
Not sure what's the intent here, but this line isn't going to trigger "< AwayTimeout" condition ever, coz changes happen only when LastActiveAt is bumped to now, right?